accname/name/comp_name_from_content.html

```diff
@@ -2,35 +2,183 @@
+<br>
+
+<h1>simple w/ for each child</h1>
+<button data-expectedlabel="one two three" data-testname="button name from content for each child" class="ex"><span>one</span> <span>two</span> <span>three</span></button><br>
+<h3 data-expectedlabel="one two three" data-testname="heading name from content for each child" class="ex"><span>one</span> <span>two</span> <span>three</span></h3>
+<a href="#" data-expectedlabel="one two three" data-testname="link name from content for each child" class="ex"><span>one</span> <span>two</span> <span>three</span></a><br>
+<br>
+
+<h1>simple for each child with image</h1>
+<button data-expectedlabel="one two three" data-testname="button name from content for each child including image" class="ex"><span>one</span> <img src="#" alt="two"> <span>three</span></button><br>
+<h3 data-expectedlabel="one two three" data-testname="heading name from content for each child including image" class="ex"><span>one</span> <img src="#" alt="two"> <span>three</span></h3>
+<a href="#" data-expectedlabel="one two three" data-testname="link name from content for each child including image" class="ex"><span>one</span> <img src="#" alt="two"> <span>three</span></a><br>
+<br>
+
+
+<h1>simple for each child with extra nesting containing image</h1>
+<button data-expectedlabel="one two three four" data-testname="button name from content for each child including nested image" class="ex"><span>one</span> <span>two <img src="#" alt="three"></span> <span>four</span></button><br>
+<h3 data-expectedlabel="one two three four" data-testname="heading name from content for each child including nested image" class="ex"><span>one</span> <span>two <img src="#" alt="three"></span> <span>four</span></h3>
+<a href="#" data-expectedlabel="one two three four" data-testname="link name from content for each child including nested image" class="ex"><span>one</span> <span>two <img src="#" alt="three"></span> <span>four</span></a><br>
+<br>
+
+<h1>heading with nested link with nested image</h1>
+<h3 data-expectedlabel="heading link image link heading" data-testname="heading name from content for each child including nested link with nested image" class="ex">
+  heading
+  <a href="#">
+    link
+    <img src="#" alt="image">
+    link
+  </a>
+  heading
+</h3>
+
+<h1>heading with nested link with nested image using aria-labelledby</h1>
+<h3 data-expectedlabel="heading image heading" data-testname="heading name from content for each child including nested link using aria-labelledby with nested image" class="ex">
+  heading
+  <a href="#" aria-labelledby="nested_image_label_1"><!-- should skip the other link text -->
+    link
+    <img id="nested_image_label_1" src="#" alt="image">
+    link
+  </a>
+  heading
+</h3>
 
-<!--
+<h1>heading with two nested links referencing image using aria-labelledby</h1>
```

Comment:
@cookiecrook Would also recommend a duplicate example with recursion depth = 3 (highly unlikely and definitely an edge case...)

Reply:
currently lines 209 to 227 below

Reply:
> I wrote the new test cases, but I hope no one ever uses something like these test cases in production.

💯